stackslib/src/net/httpcore.rs

```diff
@@ -1576,27 +1667,36 @@ impl ProtocolFamily for StacksHttp {
     ) -> Result<(), NetError> {
         match *message {
             StacksHttpMessage::Request(ref req) => {
-                // client cannot send more than one request in parallel
-                if self.request_handler_index.is_some() {
-                    test_debug!("Have pending request already");
-                    return Err(NetError::InProgress);
-                }
+                // the node cannot send more than one request in parallel, unless the client is
+                // directing it
+                let handler_index = if !self.allow_arbitrary_response {
+                    if self.request_handler_index.is_some() {
+                        test_debug!("Have pending request already");
+                        return Err(NetError::InProgress);
+                    }
 
-                // find the response handler we'll use
-                let (decoded_path, _) = decode_request_path(&req.preamble().path_and_query_str)?;
-                let handler_index = self
-                    .find_response_handler(&req.preamble().verb, &decoded_path)
-                    .ok_or(NetError::SendError(format!(
-                        "No response handler found for `{} {}`",
-                        &req.preamble().verb,
-                        &decoded_path
-                    )))?;
+                    // find the response handler we'll use
+                    let (decoded_path, _) =
+                        decode_request_path(&req.preamble().path_and_query_str)?;
+                    let handler_index = self
+                        .find_response_handler(&req.preamble().verb, &decoded_path)
+                        .ok_or(NetError::SendError(format!(
+                            "No response handler found for `{} {}`",
+                            &req.preamble().verb,
+                            &decoded_path
+                        )))?;
+                    handler_index
+                } else {
+                    0
```

Comment:
As far as I can tell, if the value here is `0` it is invalid and never used. How about using `None` here instead?

Reply:
Sure